shell/platform/darwin/common/framework/Headers/FlutterDartProject.h

```diff
@@ -6,32 +6,58 @@
 #define FLUTTER_FLUTTERDARTPROJECT_H_
 
 #import <Foundation/Foundation.h>
+#import <TargetConditionals.h>
 
 #import "FlutterMacros.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
 /**
  * A set of Flutter and Dart assets used by a `FlutterEngine` to initialize execution.
+ *
+ * TODO(stuartmorgan): Align API with FlutterDartProject, and combine.
  */
 FLUTTER_DARWIN_EXPORT
 @interface FlutterDartProject : NSObject
 
 /**
  * Initializes a Flutter Dart project from a bundle.
+ *
+ * The bundle must either contain a flutter_assets resource directory, or set the Info.plist key
+ * FLTAssetsPath to override that name (if you are doing a custom build using a different name).
+ *
+ * @param bundle The bundle containing the Flutter assets directory. If nil, the App framework
+ *               created by Flutter will be used.
  */
 - (instancetype)initWithPrecompiledDartBundle:(nullable NSBundle*)bundle NS_DESIGNATED_INITIALIZER;
-
 /**
  * Unavailable - use `init` instead.
  */
-- (instancetype)initFromDefaultSourceForConfiguration FLUTTER_UNAVAILABLE("Use -init instead.");
+- (instancetype)initFromDefaultSourceForConfiguration API_UNAVAILABLE(macos)
+    FLUTTER_UNAVAILABLE("Use -init instead.");
+
+/**
+ * Returns the default identifier for the bundle where we expect to find the Flutter Dart
+ * application.
+ */
++ (NSString*)defaultBundleIdentifier API_UNAVAILABLE(macos);
```

Comment:
The implementation of this is a single line (return [this constant](https://github.com/flutter/engine/blob/d2faa10bb784eb19b3a03737d66260d12f963b5a/shell/platform/darwin/macos/framework/Source/FlutterDartProject.mm#L13)), so it would be just as easy to make this available on macOS and avoid the extra difference.